.github/workflows/main.yml

```
@@ -66,6 +66,41 @@ jobs:
         name: engine
         path: dist
 
+  run_wpt:
+    name: Run Web Platform Tests
+    needs: [build_engine]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Download Engine
+      uses: actions/download-artifact@v1
+      with:
+        name: engine
+
+    - name: Install Viceroy
+      run: cargo install viceroy --version 0.2.4
```

This is how I would cache the cargo install, I do similar things in polyfill-library's workflows
```suggestion
      - name: add viceroy-version env
        run: echo "viceroy-version=0.2.4" >> $GITHUB_ENV
      - name: cache viceroy
        id: cache-viceroy
        uses: actions/cache@v2.1.5
        with:
          path: "/home/runner/.cargo/bin/viceroy"
          key: cache--dist--${{ env.viceroy-version }}
      - name: Install Viceroy
        run: cargo install viceroy --version 0.2.4
        if: steps.cache-viceroy.outputs.cache-hit != 'true'
```

---

Thank you, this was an excellent starting point! I ended up doing [something slightly more involved](https://github.com/fastly/js-compute-runtime/pull/40/commits/175fc53dc9fafbd1aca6dca73ee8b3a925b3e151), both because we have multiple crates to install, and because I want them to build in parallel, but it would've taken me much longer to get to a solution without this suggestion <3